flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/RowFunctionITCase.java

```diff
@@ -83,6 +92,36 @@ Stream<TestSetSpec> getTestSetSpecs() {
                                         .notNull()));
     }
 
+    @Test
+    public void testCastToSmallIntAndTinyIntIfInputIsNonLiteral() throws Exception {
+        final TableEnvironment env = TableEnvironment.create(EnvironmentSettings.inStreamingMode());
+
+        TableResult result =
+                env.executeSql(
+                        "SELECT ROW(CAST(a AS SMALLINT), CAST(b AS TINYINT)) FROM (VALUES (1, 2)) AS T(a, b)");
```

Comment:
```suggestion
                        "SELECT CAST(ROW(CAST(a AS SMALLINT), CAST(b AS TINYINT)) AS ROW<a SMALLINT, b TINYINT>) AS `row` FROM (VALUES (1, 2)) AS T(a, b)");
```
to have nicer names instead of `EXPR$`